dds/src/domain/domain_participant_factory.rs

```diff
@@ -138,7 +143,7 @@ fn configuration_try_from_str(configuration_json: &str) -> Result<DustDdsConfigu
 /// [`DomainParticipantFactory`] itself has no factory. It is a pre-existing singleton object that can be accessed by means of the
 /// [`DomainParticipantFactory::get_instance`] operation.
 pub struct DomainParticipantFactory {
-    participant_list: DdsRwLock<Vec<DcpsService>>,
+    participant_list: DdsRwLock<Vec<DdsShared<DcpsService>>>,
```

Comment:
That the dcps service needs to be shared does sound dangerous. I was hoping the service uses other things, and is not used by other things.

Reply:
I hope this is a temporary situation but it needed to be able to not have the condvar/sender inside the underlying objects so I accept it for now.